paz/pipelines/detection.py

```diff
@@ -886,3 +886,71 @@ def __init__(self, score_thresh=0.60, nms_thresh=0.45, draw=True):
                                base_weights='VOC', head_weights='VOC')
         super(EFFICIENTDETD0VOC, self).__init__(
             model, names, score_thresh, nms_thresh, draw=draw)
+
+
+class DetectVVAD(Processor):
+    """Visual Voice Activity Detection classification and detection pipeline.
+
+    # Example
+        ``` python
+        from paz.backend.camera import VideoPlayer, Camera
+        import paz.pipelines.detection as dt
+
+        detect = DetectVVAD()
+
+        pipeline = dt.DetectVVAD()
+        # To input multiple images, use a camera or a prerecorded video
+        camera = Camera(args.camera_id)
+        player = VideoPlayer((640, 480), pipeline, camera)
+        player.run()
+        ```
+
+    # Returns
+        Dictionary with ``image`` and ``boxes2D``.
+
+    # Returns
+        A function that takes an RGB image and outputs the predictions
+        as a dictionary with ``keys``: ``image`` and ``boxes2D``.
+        The corresponding values of these keys contain the image with the drawn
+        inferences and a list of ``paz.abstract.messages.Boxes2D``.
+        Note multiple images are needed to produce a prediction.
+
+    # Arguments
+        architecture: String indicating the architecture to be used for
+            classification. Options are 'VVAD-LRS3-LSTM', 'CNN2Plus1D',
+            'CNN2Plus1D_Filters' and 'CNN2Plus1D_Light'
+    """
+    def __init__(self, architecture: Architecture_Options = 'CNN2Plus1D_Light', update_rate=10, average=6,
+                 average_type: Average_Options = 'weighted', offsets=[0, 0], colors=EMOTION_COLORS):
+        super(DetectVVAD, self).__init__()
+        self.offsets = offsets
+        self.colors = colors
+
+        # detection
+        self.detect = HaarCascadeFrontalFace()
+        self.square = SequentialProcessor()
+        self.square.add(pr.SquareBoxes2D())
+        self.square.add(pr.OffsetBoxes2D(offsets))
+        self.clip = pr.ClipBoxes2D()
+        self.crop = pr.CropBoxes2D()
+
+        # classification
+        self.classify = ClassifyVVAD(update_rate=update_rate, average=average, average_type=str(average_type),
+                                     architecture=architecture)
+
+        # drawing and wrapping
+        self.class_names = self.classify.class_names
+        self.draw = pr.DrawBoxes2D(self.class_names, self.colors, True)
+        self.wrap = pr.WrapOutput(['image', 'boxes2D'])
+
+    def call(self, image):
+        boxes2D = self.detect(image.copy())['boxes2D']
+        boxes2D = self.square(boxes2D)
+        boxes2D = self.clip(image, boxes2D)
+        cropped_images = self.crop(image, boxes2D)
+        for cropped_image, box2D in zip(cropped_images, boxes2D):
+            predictions = self.classify(cropped_image)
+            box2D.class_name = predictions['class_name']
+            box2D.score = np.amax(predictions['scores'] is None if -1.0 else predictions['scores'])
```

Comment:
Maybe one could write a small backend function + processor for this operation?

Reply:
Added in [commit](https://github.com/oarriaga/paz/pull/330/commits/c95d441b681c5d2211889bf9289db0359dbc8dd0)